src/Runner.Common/ResultsServer.cs

```diff
@@ -94,5 +117,138 @@ public void InitializeResultsClient(Uri uri, string token)
 
             throw new InvalidOperationException("Results client is not initialized.");
         }
+
+        public ValueTask DisposeAsync()
+        {
+            CloseWebSocket(WebSocketCloseStatus.NormalClosure, CancellationToken.None);
+
+            GC.SuppressFinalize(this);
+
+            return ValueTask.CompletedTask;
+        }
+
+        private void InitializeWebsocketClient(string liveConsoleFeedUrl, string accessToken, TimeSpan delay)
+        {
+            if (!string.IsNullOrEmpty(accessToken))
+            {
+                if (!string.IsNullOrEmpty(liveConsoleFeedUrl))
+                {
+                    // let's ensure we use the right scheme
+                    liveConsoleFeedUrl = liveConsoleFeedUrl.Replace("https://", "wss://").Replace("http://", "ws://");
```

Comment:
do we still need to do this URL magic for URL from result service?
we were doing it due to the limitation of legacy LocationService.

Reply:
I will make server changes.